flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/SourceOperator.java

```diff
@@ -347,6 +375,7 @@ private DataInputStatus emitNextNotReading(DataOutput<OUT> output) throws Except
         switch (operatingMode) {
             case OUTPUT_NOT_INITIALIZED:
                 currentMainOutput = eventTimeLogic.createMainOutput(output);
+                latencyMarerEmitter.setMainOutput(output);
```

Comment:
@pnowojski , IMO, I don't think it's better to consruct the `latencyMarkerEmitter`. Because the watermark emitter isn't constructed here in `SourceOperator`,  the latency marker emitter should follow the pattern of the watermark emitter.

Reply:
@pnowojski , I prefer to construct the `LatencyMarkerEmitter` in the `open` method, which follows the pattern of `eventTimeLogic` and set the `DataOutput` in the `emitNext` method.